app/uk/gov/hmrc/gatekeeper/controllers/UpdateApplicationNameController.scala

```diff
@@ -112,14 +113,14 @@ class UpdateApplicationNameController @Inject() (
   def updateApplicationNameAdminEmailAction(appId: ApplicationId) = anyStrideUserAction { implicit request =>
     withApp(appId) { app =>
       def handleValidForm(form: UpdateApplicationNameAdminEmailForm) = {
-        val newApplicationName = request.session.get(newAppNameSessionKey).get
+        val newApplicationName = ValidatedApplicationName(request.session.get(newAppNameSessionKey).get).get // Already validated by the form
```

Comment:
have you got an unsafe apply you can use - better error?

Reply:
Done